storagemarket/deal_execution.go

```diff
@@ -495,10 +528,11 @@ func (p *Provider) fireEventDealUpdate(pub event.Emitter, deal *types.ProviderDe
 	}
 }
 
-func (p *Provider) updateCheckpoint(ctx context.Context, pub event.Emitter, deal *types.ProviderDealState, ckpt dealcheckpoints.Checkpoint) error {
+func (p *Provider) updateCheckpoint(pub event.Emitter, deal *types.ProviderDealState, ckpt dealcheckpoints.Checkpoint) error {
 	prev := deal.Checkpoint
 	deal.Checkpoint = ckpt
-	if err := p.dealsDB.Update(ctx, deal); err != nil {
+	// we don't want a graceful shutdown to mess with db updates
+	if err := p.dealsDB.Update(context.TODO(), deal); err != nil {
```

Comment:
```suggestion
	if err := p.dealsDB.Update(context.Background(), deal); err != nil {
```
https://www.programming-books.io/essential/go/context.todo-vs.context.background-d5224e27ff724a33a79cb4e03a5eb333.html